addons/account_edi_ubl_cii_tax_extension/models/account_edi_common.py

```diff
@@ -0,0 +1,77 @@
+    '143-1F': 'Exempt based on article 143, section 1 (f) of Council Directive 2006/112/EC',
+    '143-1FA': 'Exempt based on article 143, section 1 (fa) of Council Directive 2006/112/EC',
+    '143-1G': 'Exempt based on article 143, section 1 (g) of Council Directive 2006/112/EC',
+    '143-1H': 'Exempt based on article 143, section 1 (h) of Council Directive 2006/112/EC',
+    '143-1I': 'Exempt based on article 143, section 1 (i) of Council Directive 2006/112/EC',
+    '143-1J': 'Exempt based on article 143, section 1 (j) of Council Directive 2006/112/EC',
+    '143-1K': 'Exempt based on article 143, section 1 (k) of Council Directive 2006/112/EC',
+    '143-1L': 'Exempt based on article 143, section 1 (l) of Council Directive 2006/112/EC',
+    '148': 'Exempt based on article 148 of Council Directive 2006/112/EC',
+    '148-A': 'Exempt based on article 148, section (a) of Council Directive 2006/112/EC',
+    '148-B': 'Exempt based on article 148, section (b) of Council Directive 2006/112/EC',
+    '148-C': 'Exempt based on article 148, section (c) of Council Directive 2006/112/EC',
+    '148-D': 'Exempt based on article 148, section (d) of Council Directive 2006/112/EC',
+    '148-E': 'Exempt based on article 148, section (e) of Council Directive 2006/112/EC',
+    '148-F': 'Exempt based on article 148, section (f) of Council Directive 2006/112/EC',
+    '148-G': 'Exempt based on article 148, section (g) of Council Directive 2006/112/EC',
+    '151': 'Exempt based on article 151 of Council Directive 2006/112/EC',
+    '151-1A': 'Exempt based on article 151, section 1 (a) of Council Directive 2006/112/EC',
+    '151-1AA': 'Exempt based on article 151, section 1 (aa) of Council Directive 2006/112/EC',
+    '151-1B': 'Exempt based on article 151, section 1 (b) of Council Directive 2006/112/EC',
+    '151-1C': 'Exempt based on article 151, section 1 (c) of Council Directive 2006/112/EC',
+    '151-1D': 'Exempt based on article 151, section 1 (d) of Council Directive 2006/112/EC',
+    '151-1E': 'Exempt based on article 151, section 1 (e) of Council Directive 2006/112/EC',
+    '309': 'Exempt based on article 309 of Council Directive 2006/112/EC',
+    'AE': 'Reverse charge',
+    'D': 'Intra-Community acquisition from second hand means of transport',
+    'F': 'Intra-Community acquisition of second hand goods',
+    'G': 'Export outside the EU',
+    'I': 'Intra-Community acquisition of works of art',
+    'IC': 'Intra-Community supply',
+    'O': 'Not subject to VAT',
+    'J': 'Intra-Community acquisition of collectors items and antiques',
+    'FR-FRANCHISE': 'France domestic VAT franchise in base',
+    'FR-CNWVAT': 'France domestic Credit Notes without VAT, due to supplier forfeit of VAT for discount',
+}
+
+
+class AccountEdiCommon(models.AbstractModel):
+    _inherit = "account.edi.common"
+
+    def _get_tax_unece_codes(self, invoice, tax):
+        if tax.ubl_cii_tax_category_code and tax.ubl_cii_tax_exemption_reason_code:
```

Comment:
the exemption reason code may not be filled but we still need to report the tax category code (if the tax is not an "Exempt" one)

Reply:
How do you distinguish "Exempt" tax, technically wise?

Reply:
Is `AE - reverse charge` considered as exempt tax?

Reply:
See https://docs.peppol.eu/poacc/billing/3.0/syntax/ubl-invoice/cac-TaxTotal/cac-TaxSubtotal/cac-TaxCategory/cbc-TaxExemptionReasonCode/